sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -132,16 +134,49 @@ def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
         - custom model name (``alias``) - used from ``algorithm_name``
         """
 
+    def _get_valid_parameters(self: "_NeuralForecastAdapter") -> dict:
+        """Get valid parameters for the underlying NeuralForecast algorithm class.
+
+        Returns
+        -------
+        dict
+            valid list of arguments for the underlying algorithm class
+        """
+        from pytorch_lightning import Trainer
+
+        model_class = self.algorithm_class
+        trainer_params = list(signature(Trainer.__init__).parameters.keys())
```

Comment:
I am not sure if I follow this. Why are you checking for parameters of `Trainer` class from `pytorch-lightning`? Should you not check only for the underlying model class in `neuralforecast`? Can you please explain?

Does `neuralforecast` really accept all parameters of `Trainer` to be passed directly? If for example someone modifies `__init__` of `NeuralForecastLSTM` class with every single argument of `Trainer` clas, those will pass this check but eventually fail right? Please let me know if I am missing something obvious.

Reply:
yes yes neuralforecast models have a parameter like **trainer_kwargs which accepts all the parameter for lightning [trainer](https://nixtlaverse.nixtla.io/neuralforecast/models.lstm.html)

Reply:
Thanks for pointing that out. I completely forgot.